util/generate-rsa-signing-tests.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python2
+# Copyright 2016 Dirkjan Ochtman.
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND AND THE AUTHORS DISCLAIM ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY
+# SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
+# OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+'''
+Script to generate *ring* test file for RSA PKCS1 v1.5 signing test vectors
+from the NIST FIPS 186-4 test vectors. Takes as single argument on the
+command-line the path to the test vector file (tested with SigGen15_186-3.txt).
+
+Requires the cryptography library from pyca.
+'''
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization, hashes
+from cryptography.hazmat.primitives.asymmetric import rsa, padding
+import sys, copy
+
+def parse(fn):
+    '''Parse input test vector file, leaving out comments and empty lines, and
+    returns a list of self-contained test cases. Depends on the S key (for
+    Signature) being the last value in each test case.'''
+    cases = []
+    with open(fn) as f:
+        cur = {}
+        for ln in f:
+            if not ln.strip():
+                continue
+            if ln[0] in {'#', '['}:
+                continue
+            name, val = ln.split('=', 1)
+            cur[name.strip()] = val.strip()
+            if name.strip() == 'S':
+                cases.append(cur)
+                cur = copy.copy(cur)
+    return cases
+
+def main(fn):
+    for case in parse(fn):
+
+        if case['SHAAlg'] == 'SHA224':
+            # SHA224 not supported in *ring* for now
```

Comment:
NIT: Please end comments with periods. (We follow the BoringSSL rule where, unless constrained for space, we make comments full sentences.) Here and elsewhere below.